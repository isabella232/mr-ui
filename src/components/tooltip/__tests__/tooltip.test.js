import React from 'react';
import { shallow } from 'enzyme';
import { testCases } from './tooltip-test-cases';

describe('Tooltip', () => {
  let testCase;
  let wrapper;

  describe(testCases.basic.description, () => {
    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(testCases.allProps.description, () => {
    beforeEach(() => {
      testCase = testCases.allProps;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
