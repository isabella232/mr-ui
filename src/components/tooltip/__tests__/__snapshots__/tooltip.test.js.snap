// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip all props renders 1`] = `
<PopoverTrigger
  content={[Function]}
  disabled={true}
  display="inline"
  popoverProps={
    Object {
      "accessibleTitle": null,
      "alignment": "left",
      "backgroundColor": "red",
      "contentElementAttributes": Object {
        "id": "tooltip-2",
        "role": "tooltip",
      },
      "hasPointer": true,
      "hideWhenAnchorIsOffscreen": true,
      "placement": "top",
      "themePopover": "py6 px12 round txt-s txt-bold",
    }
  }
  receiveFocus={false}
  renderHiddenContent={true}
  respondsToClick={true}
  respondsToFocus={true}
  respondsToHover={true}
  trapFocus={false}
  triggerProps={Object {}}
>
  <span>
    test tooltip
  </span>
</PopoverTrigger>
`;

exports[`Tooltip basic renders 1`] = `
<PopoverTrigger
  content="basic"
  disabled={false}
  display="inline-block"
  popoverProps={
    Object {
      "accessibleTitle": null,
      "alignment": "center",
      "backgroundColor": "#fff",
      "contentElementAttributes": Object {
        "id": "tooltip-1",
        "role": "tooltip",
      },
      "hasPointer": true,
      "hideWhenAnchorIsOffscreen": true,
      "placement": "top",
      "themePopover": "round py6 px12 shadow-darken25 txt-s z1",
    }
  }
  receiveFocus={false}
  renderHiddenContent={true}
  respondsToClick={false}
  respondsToFocus={true}
  respondsToHover={true}
  trapFocus={false}
  triggerProps={Object {}}
>
  <span>
    test tooltip
  </span>
</PopoverTrigger>
`;
